controllers/tasks.js

```diff
@@ -51,24 +52,29 @@ router.get('/cron', (req, res, next) => {
 });
 
 /**
- * GET /tasks/execute
+ * GET /tasks/execute?tasks=1
  *
  * We use a GET from the cron job to execute each PWA update task
+ * The tasks parameter is the number of tasks to execute per run
  */
 router.get('/execute', (req, res, next) => {
   // Checks for the presence of the 'X-Appengine-Cron' header on the request.
   // Only requests from the App Engine cron are allowed.
+  const tasksToExecute = req.query.tasks ? req.query.tasks : 1;
+  const tasksList = [];
   if (req.get(APP_ENGINE_CRON)) {
-    tasksLib.pop()
-      .then(task => {
-        if (task) {
-          tasksLib.executePwaTask(task);
-        }
-        res.sendStatus(200);
-      })
-      .catch(err => {
-        next(err);
-      });
+    try {
+      for (var i = 0; i < tasksToExecute; i++) {
+        tasksList.push(tasksLib.popExecute);
+      }
+      // Execute sequentially the 1 OR req.query.tasks of tasks
+      promiseSequential.all(tasksList)
```

Comment:
@andreban it will successfully run all until one fails, it will return an error for the cron job and log errors.

Each task has a retry mechanism, so they will be queued again, the ones after the one that fails will not be pop out of the queue, so the next cron will pick them up.